Development/nmos/streamcompatibility_api.h

```diff
@@ -0,0 +1,45 @@
+#ifndef NMOS_STREAMCOMPATIBILITY_API_H
+#define NMOS_STREAMCOMPATIBILITY_API_H
+
+#include <boost/variant.hpp>
+#include "bst/optional.h"
+#include "nmos/api_utils.h"
+#include "nmos/slog.h"
+
+// Stream Compatibility Management API implementation
+// See https://specs.amwa.tv/is-11/branches/v1.0-dev/APIs/StreamCompatibilityManagementAPI.html
+namespace nmos
+{
+    struct node_model;
+    struct resource;
+
+    namespace experimental
+    {
+        namespace details
+        {
+            // a streamcompatibility_base_edid_handler is a notification that the Base EDID for the specified IS-11 input has changed (PUT or DELETEd)
+            // it can be used to perform any final validation of the specified Base EDID and set parsed Base EDID properties
+            // when PUT, it may throw web::json::json_exception, which will be mapped to a 400 Bad Request status code with NMOS error "debug" information including the exception message
+            // or std::runtime_error, which will be mapped to a 500 Internal Error status code with NMOS error "debug" information including the exception message
+            // when DELETE, this callback should not throw exceptions, as the Base EDID will already have been changed and those changes will not be rolled back
+            typedef std::function<void(const nmos::id& input_id, const bst::optional<utility::string_t>& base_edid, web::json::value& base_edid_properties)> streamcompatibility_base_edid_handler;
+
+            // a streamcompatibility_active_constraints_put_handler is a notification that the Active Constraints for the specified IS-11 sender has changed
+            // it can be used to perform any final validation of the specified Active Constraints
+            // it may throw web::json::json_exception, which will be mapped to a 400 Bad Request status code with NMOS error "debug" information including the exception message
+            // or std::logic_error, which will be mapped to a 422 Unprocessible Entity status code with NMOS error "debug" information including the exception message
+            // or std::runtime_error, which will be mapped to a 500 Internal Error status code with NMOS error "debug" information including the exception message
```

Comment:
In most nmos-cpp API implementations, `std::logic_error` is mapped to general `500` Internal Error and `std::runtime_error` is mapped to `501` Not Implemented. The Connection API purposefully maps both to `500`, so the behaviour for `streamcompatibility_base_edid_handler` matches...

However, using `std::logic_error` to indicate `422` could be a bit confusing because that has always mapped to `500` everywhere. Not sure whether we could use a `bool` result, or `intersection` returned as `null`, to indicate "not capable", maybe??

(BTW, the latest HTTP specs seem to spell it "Unprocessable Content" rather than "Unprocessible Entity"?)

https://github.com/sony/nmos-cpp/blob/a0ff5f372cd828e87e6ccbfa18aa0f35b01a3c2f/Development/nmos/api_utils.cpp#L637-L648